File: src/dataset/javascript/simple-cipher_exercism-javascript-bearguns/openAI/ChatGPT4_simple-cipher_v4.js
class Cipher {
  constructor(key) {
    if (key && /[^a-z]/.test(key)) {
      throw new Error('Bad key');
    }

    this.alphabet = 'abcdefghijklmnopqrstuvwxyz';
    this.key = key || this.keyGen();
  }

  keyGen() {
    return Array.from({ length: 100 }, () =>
      this.alphabet.charAt(Math.floor(Math.random() * 26))
    ).join('');
  }

  encode(plaintext) {
    return plaintext
      .split('')
      .map((char, i) => {
        const shift = this.alphabet.indexOf(this.key[i % this.key.length]);
        const newIndex = (this.alphabet.indexOf(char) + shift) % 26;
        return this.alphabet[newIndex];
      })
      .join('');
  }

  decode(ciphertext) {
    return ciphertext
      .split('')
      .map((char, i) => {
        const shift = this.alphabet.indexOf(this.key[i % this.key.length]);
        const newIndex = (this.alphabet.indexOf(char) - shift + 26) % 26;
        return this.alphabet[newIndex];
      })
      .join('');
  }
}

export default Cipher;

File: src/dataset/Java/simple-cipher_exercism-java-ThomasZumsteg/openAI/ChatGPT4_simple-cipher_v4.java
import java.util.Random;

public class Cipher {
    private final int[] key;

    public Cipher() {
        Random random = new Random();
        key = random.ints(100, 0, 'z' - 'a').toArray();
    }

    public Cipher(String key) {
        if (key.isEmpty() || !key.chars().allMatch(c -> c >= 'a' && c <= 'z')) {
            throw new IllegalArgumentException();
        }
        this.key = key.chars().map(c -> c - 'a').toArray();
    }

    public String getKey() {
        char[] keyChars = new char[key.length];
        for (int i = 0; i < key.length; i++) {
            keyChars[i] = (char) (key[i] + 'a');
        }
        return new String(keyChars);
    }

    public String encode(String plainText) {
        char[] cipherText = new char[plainText.length()];
        for (int i = 0; i < plainText.length(); i++) {
            cipherText[i] = shift(plainText.charAt(i), key[i % key.length]);
        }
        return new String(cipherText);
    }

    public String decode(String cipherText) {
        char[] plainText = new char[cipherText.length()];
        for (int i = 0; i < cipherText.length(); i++) {
            plainText[i] = shift(cipherText.charAt(i), -key[i % key.length]);
        }
        return new String(plainText);
    }

    private char shift(char c, int k) {
        int s = c + k;
        if (s < 'a') s += 'z' - 'a' + 1;
        else if (s > 'z') s -= 'z' - 'a' + 1;
        return (char) s;
    }
}